doc/templates.rst

```diff
@@ -76,8 +76,24 @@ Twig templates have access to variables provided by the PHP application and
 variables created in templates via the :doc:`set <tags/set>` tag. These
 variables can be manipulated and displayed in the template.
 
-Use a dot (``.``) to access attributes of a variable (methods, properties
-or constants of a PHP object, or items of a PHP array):
+Twig tries to abstract PHP types as much as possible and works with a few basic
+types, supported by ``filters``, ``functions``, and ``tests`` among others:
+
+===================  ===============================
+Twig Type            PHP Type
+===================  ===============================
+string               A string or a Stringable object
+number               An integer or a float
```

Comment:
In the docs `int` is used here: https://twig.symfony.com/doc/3.x/tags/types.html#types
Do we specify for the new feature PHP types instead of Twig types?

Reply:
Good catch!

While the documentation states "The syntax for and contents of type strings are intentionally left out of scope", I do agree that "number" and "boolean" are better choices to align with the documentation on types.

I'll create a PR for updating the `types` documentation.

Reply:
Voila: #4403